<!DOCTYPE html>
<html lang="en-US">


 <!-- Begin Jekyll SEO tag v2.8.0 -->
<title>BiG-SCAPE / CORASON tutorial | BiG-SCAPE CORASON</title>
<meta name="generator" content="Jekyll v3.9.3" />
<meta property="og:title" content="BiG-SCAPE / CORASON tutorial" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="July 2018" />
<meta property="og:description" content="July 2018" />
<link rel="canonical" href="http://localhost:4000/bigscape-corason/tutorial/" />
<meta property="og:url" content="http://localhost:4000/bigscape-corason/tutorial/" />
<meta property="og:site_name" content="BiG-SCAPE CORASON" />
<meta property="og:type" content="website" />
<meta name="twitter:card" content="summary" />
<meta property="twitter:title" content="BiG-SCAPE / CORASON tutorial" />
<script type="application/ld+json">
{"@context":"https://schema.org","@type":"WebPage","description":"July 2018","headline":"BiG-SCAPE / CORASON tutorial","url":"http://localhost:4000/bigscape-corason/tutorial/"}</script>
<!-- End Jekyll SEO tag -->

  </head>

  <body>


    <!-- HEADER -->
    <div id="header_wrap" class="outer">
        <header class="inner">
          
    <head>


	<meta charset="utf-8">
	<meta http-equiv="x-ua-compatible" content="ie=edge">
	<meta name="viewport" content="width=device-width, initial-scale=1">

	<title>BiG-SCAPE / CORASON tutorial</title>

	<link rel="stylesheet" href="/bigscape-corason/css/main.css">
	<link rel="canonical" href="http://localhost:4000/bigscape-corason/tutorial/">
	<link rel="shortcut icon" href="/bigscape-corason/images/favicon.png"/>
	<link rel="stylesheet" type="text/css" href="/bigscape-corason/css/formstyle2.css"  />

	<link href='https://fonts.googleapis.com/css?family=Open+Sans:400,300,700,800,600' rel='stylesheet' type='text/css'>
	<link href='https://fonts.googleapis.com/css?family=Muli:400,300' rel='stylesheet' type='text/css'>
	<link type="text/css" href="/bigscape-corason/css/jquery-ui-1.10.4.custom.css" rel="stylesheet" />
	<link type="text/css" href="/bigscape-corason/css/alpaca_custom.css" rel="stylesheet" />
	<link type="text/css" href="/bigscape-corason/css/alpaca-jqueryui.css" rel="stylesheet" />

</head> 

        </header>
    </div>


 <div id="page-wrap">
    <!-- MAIN CONTENT -->
    <div id="main_content_wrap" class="outer">
      <section id="main_content" class="inner">
	 


   <div id="form-header">
      <div id="logo1">
      <img src="/bigscape-corason/images/bigscape.png" width="23%" align="left">
      </div>
      <div id="logo2">
      <img src="/bigscape-corason/images/corason.png" width="175" height="86" align="right">
   
      </div>
      <div id="menu">
        <a href=/bigscape-corason/>HOME</a>
        &nbsp;&nbsp;|&nbsp;&nbsp;
        <a href="/bigscape-corason/installation">INSTALLATION</a>
        &nbsp;&nbsp;|&nbsp;&nbsp;
        <a href="/bigscape-corason/tutorial">TUTORIAL</a>
        &nbsp;&nbsp;|&nbsp;&nbsp;
        <a href="/bigscape-corason/about">ABOUT</a>
        &nbsp;&nbsp;|&nbsp;&nbsp;
        <br><br>
    
      </div>
      </div>      
    <hr>

        <h2>BiG-SCAPE / CORASON tutorial</h2>
   
    <p>This tutorial explains how to download datasets to run BiG-SCAPE and CORASON, and provides examples for how to use both tools.</p>

<h2 id="example-dataset">Example dataset</h2>

<p>The example dataset consists on three parts: a set of genomes (includes a single cluster from MIBiG), a set of GenBank files (selected clusters predicted from the set of genomes) and the TauD sequence from a <i>Streptomyces</i> genome (NRRL B-1347).</p>

<p>Download and execute the following script to obtain the files:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">mkdir</span> ~/bin/example    <span class="c"># not required if you already have that</span>
<span class="nv">$ </span>curl https://raw.githubusercontent.com/nselem/bigscape-corason/master/scripts/data_bigscape_corason.sh
<span class="nv">$ </span><span class="nb">mv </span>data_bigscape_corason.sh ~/bin/example/data_bigscape_corason.sh
<span class="nv">$ </span><span class="nb">chmod </span>a+x ~/bin/example/data_bigscape_corason.sh
<span class="nv">$ </span><span class="nb">cd</span> ~/bin/example <span class="o">&amp;&amp;</span> ~/bin/example/data_bigscape_corason.sh
</code></pre></div></div>

<p>The content of the <code class="language-plaintext highlighter-rouge">~/bin/example</code> folder should look like this:</p>

<center>
    <img src="/bigscape-corason/images/contents_example_data_package.png" alt="Content of the package with example data" />
</center>

<p>Data can also be downloaded manually at: <a href="https://doi.org/10.5281/zenodo.1340706"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.1340706.svg" alt="Link to the Zenodo repository with the example data" /></a></p>

<h3> How to compile your own input dataset </h3>

<p>In this tutorial, we will work with the above example data. If you want to use your own gene clusters as input for BiG-SCAPE and the CORASON family mode integrated within the BiG-SCAPE pipeline, you can search for publicly available genomes in
<a href="https://antismash-db.secondarymetabolites.org/#!/start">antiSMASH-DB</a> and download the desired cluster files in GenBank (.gbk) format. Alternatively, you can perform your own antiSMASH runs on the public <a href="https://antismash.secondarymetabolites.org/#!/start">web server</a> or on your local system, collect the cluster GBK files and put them together in a folder that you can use as input for BiG-SCAPE. Entries from MIBiG can be added automatically by adding the <code class="language-plaintext highlighter-rouge">--mibig</code> flag to the end of your BiG-SCAPE command (see below).</p>

<h2 id="big-scape-example">BiG-SCAPE example</h2>

<p>We will now proceed with the example data. Once data has been downloaded, run the following command at the terminal (you should be now in the <code class="language-plaintext highlighter-rouge">~/bin/example</code> folder):</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>run_bigscape gbks example_output
</code></pre></div></div>

<p>If everything goes well, the terminal will show something similar to:</p>

<center>
    <img src="/bigscape-corason/images/bigscape_example1.png" alt="BiG-SCAPE example terminal" />
</center>

<p>The input for BiG-SCAPE is the directory gbks that contains GenBank files of sequences of Biosynthetical gene clusters (BGCs) predicted by antiSMASH. The BiG-SCAPE output will be stored in the directory <code class="language-plaintext highlighter-rouge">example_output</code>.</p>

<p>After BiG-SCAPE has finished successfully open the <code class="language-plaintext highlighter-rouge">index.html</code> file located inside the <code class="language-plaintext highlighter-rouge">example_output</code> folder with your browser, (e.g. Chrome or Firefox). The file contains an interactive offline webpage that displays the BiG-SCAPE results and allows you to explore them.</p>

<center>
    <img src="/bigscape-corason/images/bigscape_example2.png" alt="Select class" width="900" border="3px" />
</center>

<p>To start exploring, select a class at the top of the site:</p>

<center>
    <img src="/bigscape-corason/images/bigscape_example3.png" alt="Select biosynthetic class" width="500" border="3px" />
</center>

<p>Now, the screen will display a network visualization of BGC families within this class.</p>

<center>
    <img src="/bigscape-corason/images/bigscape_example4.png" alt="Network view" width="600" border="3px" />
</center>

<p>In this case, the NRPS class contains 10 BGCs organized in one gene cluster family of three members, one family of two members and five singletons.</p>

<p>Now select a family in this network to visualize BGCs sorted and aligned CORASON-like.</p>

<center>
    <img src="/bigscape-corason/images/bigscape_example5.png" alt="CORASON-like output" width="800" border="3px" />
</center>

<p>This family contains three members.</p>

<p>BiG-SCAPE output (.network files) can also be imported with Cytoscape.</p>

<p>In another example BiG-SCAPE was employed to calculate BGC familiess in 103 complete Streptomyces genomes. The outcome of this run can be found <a href="https://bigscape-corason.secondarymetabolites.org/streptomyces_example/">here</a></p>

<h2 id="corason-example">CORASON example</h2>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span> ~/bin/run_corason <span class="nt">-q</span> queryFile gbksDirectory referenceBGC <span class="nt">-g</span>
</code></pre></div></div>

<p>After getting the example data, the next step to identify variations in the genomic vicinity of <em>tauD</em> that remain similar to the reference BGC <code class="language-plaintext highlighter-rouge">JMGX01000001.1.cluster003.gbk</code> is to execute CORASON:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>~/bin/run_corason TauD.fasta gbks gbks/JMGX01000001.1.cluster003.gbk
</code></pre></div></div>

<p>The output will be in the directory query-output, in this case <code class="language-plaintext highlighter-rouge">TauD.fasta-output</code>. See the svg output (e.g. <code class="language-plaintext highlighter-rouge">$ firefox TauD.fasta-output/Joined.svg</code>). In this case we can see even fragment BGCs without the NRPS genes:</p>

<center>
    <img src="/bigscape-corason/images/corason_example.png" alt="CORASON example" height="350" border="3px" />
</center>

<p>To obtain more variations of genomic vicinities, switch the exploration to genomes instead of BGCs. Here, more BGCs are found and differences are not limited to gene content of the BGC; differences in sequence similarity are also indicated by a color gradient.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>~/bin/run_corason TauD.fasta genomes genomes/JOBW01.gbk
</code></pre></div></div>

<center>
    <img src="/bigscape-corason/images/corason_genomes.png" alt="Corason example" width="800" border="3px" /> <br />
</center>


      </section>
    </div>


    <!-- FOOTER  -->
    <div id="footer_wrap" class="outer">
      <footer class="inner">
        
        <p>Published with <a href="https://pages.github.com">GitHub Pages</a></p>
      </footer>
    </div>

    
	<main>
		<footer>
    <div id="footer"> Comments or questions on bigscape/corason? Please do not hesitate to <a href="mailto:marnix.medema@wur.nl">contact us</a>.
      <br><br><a rel="license" href="http://creativecommons.org/licenses/by/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
      </div>
</footer>


	</main>
     </div>
  </body>
</html>
